src/main/java/org/terasology/gooeyDefence/EnemyManager.java

```diff
@@ -85,6 +108,39 @@ public void onPeriodicActionTriggered(PeriodicActionTriggeredEvent event, Entity
         }
     }
 
+    /**
+     * Called when the a path is changed.
+     *
+     * @see OnPathChanged
+     */
+    @ReceiveEvent
+    public void onPathChanged(OnPathChanged event, EntityRef shrineEntity) {
+        for (EntityRef enemy : enemies) {
+            /* Firstly check if the enemy is on an unchanged path */
+            if (enemy.hasComponent(EntrancePathComponent.class)) {
```

Comment:
could you explain in what scenarios the entrance id of the path may change for an enemy?

Reply:
The entrance id of the path would never change for an enemy.
That check is to make sure that only the enemies on the path that was changed are updated.
Else if we have three paths, `0`, `1` & `2` and path `1` is changed then enemies on *all* paths will be updated. However as paths `0` & `2` are unchanged then we can skip those enemies in this method.